WalletWasabi/WabiSabi/Client/Decomposer.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+/// <summary>
+/// https://github.com/lontivero/DecompositionsPlayground/blob/master/Notebook.ipynb
+/// </summary>
+public static class Decomposer
+{
+	public static long[] StdDenoms;
+
+	public static IEnumerable<(long Sum, int Count, ulong Decomposition)> Decompose(long target, long tolerance, int maxCount)
+	{
+		if (maxCount is <= 1 or > 8)
+		{
+			throw new ArgumentOutOfRangeException(nameof(maxCount), "The maximum decomposition lenght cannot be greater than 8 or smaller than 1.");
+		}
+		if (target <= 0)
+		{
+			throw new ArgumentException("Only possitive numbers can be decomposed.", nameof(target));
+		}
+
+		var denoms = StdDenoms.SkipWhile(x => x > target).ToArray();
+
+		if (denoms.Length > 255)
+		{
+			throw new ArgumentException("Too many denominations. Maximum number is 256.", nameof(target));
```

Comment:
```suggestion
			throw new ArgumentException("Too many denominations. Maximum number is 255.", nameof(target));
```

Reply:
Yes. In fact you can have from 0 to 255 so, those are 256 but it makes no sense to have 0 denominations.